### ISSUE TITLE
`@OptimisticLocking(DIRTY)` leads to wrong query during delete of circular reference 

### ISSUE THREAD

**Issue by mickroll** [quarkusio/quarkus#24842]
### Describe the bug

Having a circular reference between two Entities and deleting one of them with cascading enabled, the final delete query fails, if the "root"-Entity uses `@OptimisticLocking(DIRTY)`.

### Expected behavior

The following queries are generated and executed:
```
update `example_entity` set `last_log_entry_id`=null where `id`=?
delete from `log_entry` where `id`=?
delete from `example_entity` where `id`=? and `last_log_entry_id` IS NULL
```
All query-parameters are filled with non-null, existing values. 
Both deletes are successful.

### Actual behavior

The following queries are generated and executed:
```
update `example_entity` set `last_log_entry_id`=null where `id`=?
delete from `log_entry` where `id`=?
delete from `example_entity` where `id`=? and `last_log_entry_id`=?
```
All query-parameters are filled with non-null, existing values.
The last query is wrong and references old state that was changed in the first query.
This leads to the following error:
```
javax.persistence.OptimisticLockException: Batch update returned unexpected row count from update [0]; actual row count: 0; expected: 1
```

### How to Reproduce?

see https://github.com/mickroll/quarkus-demonstrator/tree/optimistic-locking-circular-reference-delete
Demonstrator uses Quarkus 2.6.2.Final, but the same Bug also appears on Quarkus 2.8.0.Final.

### Output of `uname -a` or `ver`

Windows 10

### Output of `java -version`

openjdk version "11.0.12"

### Quarkus version or git rev

2.6.2.Final / 2.8.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

**Comment by Sanne** [comment]
hi @mickroll - thanks for the report. May I assume I can ask the Hibernate ORM team to have a look, or is there a specific reason to have reported it to Quarkus?

**Comment by mickroll** [comment]
As you are asking, yes this seems hibernate-related. Please feel free to forward.

**Comment by famod** [comment]
Hey @Sanne, is there a HHH issue already?

**Comment by mickroll** [comment]
FYI @Sanne i just reported this as a hibernate issue: https://hibernate.atlassian.net/browse/HHH-15218

**Comment by famod** [comment]
Hi @Sanne @yrodiere, any chance to fix this upstream? 🙂

**Comment by Sanne** [comment]
sorry for the lack of communications.

I've pinged the ORM team to make sure it gets triaged, seems to have fallen through the cracks. Not sure if we'll have the bandwith to fix it in 5, but let's see what the team thinks.

**Comment by Sanne** [comment]
this was fixed by @dreab8 in Hibernate ORM, so we'll have a fix when merging #26625

**Comment by famod** [comment]
Awesome, thanks @dreab8 & @Sanne!

**Comment by mickroll** [comment]
> this was fixed by @dreab8 in Hibernate ORM, so we'll have a fix when merging #26625

Sounds great :rocket:
